src/deployment/linux.md

```diff
@@ -305,7 +277,7 @@ Place the .desktop file in your Flutter project
 under `<project root>/snap/gui/super-cool-app.desktop`.
 
 **Notice**: icon and .desktop file name must be the 
-same as your app name in yaml file!
+same as your app name in yaml file! Also, your icon file must be world readable.
```

Comment:
It's also probably not worth calling out the permissions of the icon file, that has to be a pretty extreme corner case to try to use an icon without read permissions.

Reply:
I was annoyed that snapcraft didn't just change the permissions on the icon file during the build process. It might be an edge case, but I hit the edge.

Reply:
> I was annoyed that snapcraft didn't just change the permissions on the icon file during the build process. It might be an edge case, but I hit the edge.

Yeah, but the same could be true for lots of the files, like the desktop file and other assets.  I don't think it's worth calling all of those out, the usual mode for files created would work.

Reply:
@kenvandine , in my case, I have my umask set to 0007, a holdover from when my Unix development computer was shared by many people. The icon file was the only issue that was an issue for me. But, I agree that it is an edge case and am okay either way. Only very paranoid or very old linux programmers set their umask to 0007. I am not very paranoid. :-)